### PR TITLE
Сorrected typos, fixed ineffassign, gofmt, go_vet

### PR DESCRIPTION
**Pull request by trigun117**

### CODE REVIEW

File: solvers.go [hunk 1]
```diff
@@ -1160,13 +1160,13 @@ func (s *BarzilaiBorweinSolver) Step(model Nodes) error {
 					contractionAxes[axis] = axis
 				}
 
-				valGradDiffscalarProd, err := tensor.Contract(valueDiff, gradDiff, contractionAxes, contractionAxes)
+				valGradDiffscalarProd, _ := tensor.Contract(valueDiff, gradDiff, contractionAxes, contractionAxes)
```
Comment (chewxy) [inline solvers.go]: This is not a good idea. @siquus how should errrors be handled for Barzilai Borwein. I'm going to guess the default is to return the error?

**Comment by chewxy** [comment]
Thanks for this. More importantly thanks for finding a potential bug

I have left a comment waiting @siquus' reply on how to handle errors. If he doesn't respond in 12 hrs, can you please fix it such that the err is not shadowed?

**Comment by trigun117** [comment]
@chewxy i will check what i can do if he not respond

**Comment by chewxy** [comment]
Thanks :+1:  I'll merge this when ready.

**Comment by siquus** [comment]
@trigun117 nice catch ...somebody... forgot to actually react to err ;-)
Failing silently at that point will be very hard to debug, so if you could please add a `return errors.new` or so along the lines "operationError, Contracting value / gradient difference" that would be nicer.
Thanks!

**Comment by trigun117** [comment]
@chewxy updated code

**Comment by chewxy** [comment]
LGTM. Merged

**Pull request merged**